### PR TITLE
feat: streaming

### PR DESCRIPTION
**Pull request by debajyoti-truefoundry**

### CODE REVIEW

File: async_processor/sidecar/sidecar.py [hunk 1]
```diff
@@ -10,21 +11,79 @@
...
+            if self._stream_reader.at_eof():
+                self._batch.is_eof = True
+                break
+
+    def _reset_batch(self):
+        self._batch = StreamBatch(data="")
+        self._len = 0
+
+    async def iter(self) -> AsyncIterator[StreamBatch]:
+        while True:
```
Comment (debajyoti-truefoundry) [inline async_processor/sidecar/sidecar.py:67]: The dynamics between `_fetcher` and `iter` are hard to reason about. Add some comments to explain the expected behaviour.

Comment (debajyoti-truefoundry) [reply async_processor/sidecar/sidecar.py:67]: done

File: async_processor/worker.py [hunk 2]
```diff
@@ -198,72 +209,142 @@ async def _publish_response(
...
+        return
+    if isinstance(maybe_generaotr, GeneratorType):
+        warnings.warn(
+            "Returning an Generator from the Processor.process method "
+            "is an experimental feature, and behaviours may change later.",
+            UserWarning,
+            stacklevel=1,
+        )
+        for obj in maybe_generaotr:
+            yield obj
```
Comment (chiragjn) [inline async_processor/worker.py:231]: Wouldn't this be blocking? We don't want to convert this to async iterator using threaded implementation?

Comment (chiragjn) [reply async_processor/worker.py:231]: Also minor, typo
```suggestion
        for obj in maybe_generator:
            yield obj
```

**Review by chiragjn — APPROVED** [review]

**Review by chiragjn — APPROVED** [review]

**Pull request merged**